ddp-workspace/projects/ddp-sdk/src/lib/services/activity/activitySuggestionBuilder.service.ts

```diff
@@ -107,52 +121,32 @@ export class ActivitySuggestionBuilder {
             }
         };
 
-        let suggestionsLeft = suggestions.filter(hasAnyMatch);
-
-        const startsWithSuggestions = suggestionsLeft.reduce((accumulator: Array<TextSuggestion>, suggestion: string) => {
-            if (hasStartWithMatch(suggestion)) {
-                accumulator.push({
-                    value: suggestion,
-                    matches: [{
-                        offset: 0,
-                        length
-                    }]
-                });
-            }
-            return accumulator;
-        }, []);
-
-        suggestionsLeft = suggestionsLeft.filter((suggestion: string) => !hasStartWithMatch(suggestion));
-
-        const startOfWordSuggestions = suggestionsLeft.reduce((accumulator: Array<TextSuggestion>, suggestion: string) => {
-            const offset = hasStartOfWordMatch(suggestion);
-            if (offset !== -1) {
-                accumulator.push({
-                    value: suggestion,
-                    matches: [{
-                        offset,
-                        length
-                    }]
-                });
-            }
-            return accumulator;
-        }, []);
-
-        suggestionsLeft = suggestionsLeft.filter((suggestion: string) => hasStartOfWordMatch(suggestion) === -1);
+        let nonMatches: Array<string> = [];
+        const findMatches = (suggestionsLeft: Array<string>, matcher): Array<TextSuggestion> => {
+            nonMatches = [];
+            return suggestionsLeft.reduce((accumulator: Array<TextSuggestion>, suggestion: string) => {
+                const offset = matcher(suggestion);
+                if (offset >= 0) {
+                    accumulator.push({
+                        value: suggestion,
+                        matches: [{
+                            offset,
+                            length
+                        }]
+                    });
+                } else {
+                    nonMatches.push(suggestion);
+                }
+                return accumulator;
+            }, []);
+        }
 
-        const suggestionsWithinWords = suggestionsLeft.reduce((accumulator: Array<TextSuggestion>, suggestion: string) => {
-            const offset = indexOfFirstMatch(suggestion);
-            if (offset !== -1) {
-                accumulator.push({
-                    value: suggestion,
-                    matches: [{
-                        offset,
-                        length
-                    }]
-                });
-            }
-            return accumulator;
-        }, []);
+        let suggestionsLeft = suggestions.filter(hasAnyMatch);
+        const startsWithSuggestions = findMatches(suggestionsLeft, indexOfStartWithMatch);
+        suggestionsLeft = nonMatches;
+        const startOfWordSuggestions = findMatches(suggestionsLeft, indexOfStartOfWordMatch);
+        suggestionsLeft = nonMatches;
+        const suggestionsWithinWords = findMatches(suggestionsLeft, indexOfFirstMatch);
 
```

Comment:
The sorting and filtering here will give suggestions that look like this:

![tb-inst-auto](https://user-images.githubusercontent.com/5713833/104057352-038bdb00-51c0-11eb-8ecb-04a69109ad6b.png)